topics/genome-annotation/tutorials/hpc-for-lsgc/slides.html

```diff
@@ -0,0 +1,495 @@
+ - Evolution (Where do we come from? And other species?)
+ - Structure (What are we made of? How does our body work?)
+ - Functionality (How do changes in our DNA affect us?)
+
+- Widely used in several fields!
+ - Whole genome alignment, gene prediction, phylogenomics, pharmacogenetics, metagenomics, etc.
+
+---
+
+### What are the challenges?
+
+- A lot of progress has taken place since the beginning (around 1960s)
+- So what is the problem?
+
+ - New technology produces huge amounts of data
+ - Optimal sequence comparison has complexity O(n<sup>2</sup>), which is a lot when sequences are long
+ - Several ways to compare sequences, each having its advantages: DP-based, gapped and ungapped, alignment-free
+ - DNA contains lots of repeats and low complexity regions, which make it difficult to find the conserved signal
+ - And also makes it very hard for computational algorithms
+
+---
+
+### Why is sequence comparison hard?
+
+- The complexity of the problem grows as the length of the sequences becomes larger
+- That means that for small sequences it is OK; but for large sequences the problem can take a lot of time
+- Computing power doubles every two years according to Moore’s law but sequence databases grow purely exponential every year
+
+
+Sequence type    | Approx. length
+---        | ---          
+Gene			|  ~1,000 to 10,000 bp
+Bacteria genome		|  ~10<sup>6</sup> bp
+Mammalian chr.		|  ~10<sup>8</sup> bp
+Mammalian genome	|  ~2 * 10<sup>9</sup> bp
+Plant genome		|  ~3 * 10<sup>9</sup> bp
+Large plant genomes	|  ~2 * 10<sup>10</sup> bp
+Amoeba genome		|  ~6 * 10<sup>11</sup> bp
+
+
+---
+### Exponential growth of the Sequence Read Archive
+
+
+![](../../images/hpc-for-lsgc/SRA.png)
+
+- Other databases have experienced similar growth too
+
+---
+
+### Methods for sequence comparison
+
+- Dot-matrix comparison (no alignments)
+ - Comparing all letters with each other (extremely slow/noisy)
+
+- Alignment-free methods
+ - Similar to the previous but typically based on k-mers (words)
+
+- Aligning methods
+   - Exhaustive dynamic programming algorithms (Slow but optimal)
+        - Local alignment
+        - Global alignment
+   - Seed-based heuristics
+        - Words of fixed size are used to start alignments (Fast but less sensitivity). Can also include gaps.
+
+---
+
+### Dot-matrix methods
+
+
+- Takes each letter in both sequences and compare it to all the rest
+ - Takes quadratic time and space!
+- Does not produce alignments, but rather just represents a comparison
+- The **x** and **y** axis represent the sequences being compared
+
+![](../../images/hpc-for-lsgc/Dot01.png)
+
+---
+
+### Comparing genomes with dotplots
+
+![](../../images/hpc-for-lsgc/Dot02.png)
+
+---
+
+### How useful are raw dotplots?
+
+.pull-left[
+
+
+
+- Dot matrix methods do not take into account the global information or context
+
+ - Noise
+
+ - Where is the signal?
+]
+
+.pull-right[.image-70[![](../../images/hpc-for-lsgc/Dot03.png)]]
```

Comment:
Images Dot03.png, Dot04.png, and Dot05.png need to be resized in order to fit properly in the slides.

Reply:
I have added an extra border on these ones so that they dont stick out so much to the right. Is there a way to do it better?

Reply:
I think that now it is fine :)